j2k/src/org/jetbrains/kotlin/j2k/ExpressionConverter.kt

```diff
@@ -538,6 +538,13 @@ class DefaultExpressionConverter : JavaElementVisitor(), ExpressionConverter {
         val qualifier = expression.qualifierExpression
 
         var identifier = Identifier.withNoPrototype(referenceName, isNullable)
+
+
+        if (expression.getContainingClass()?.getParentOfType<PsiField>(false).let { it != null && it == expression.qualifierExpression?.reference?.resolve() }) {
```

Comment:
It is used in if check
Which better
`if(x.let{ it != null && ........................long_check.....................................})`
or
`if(x?.let{  ........................long_check..................................... } ?: false)`